shayan / cashier documentation updated (#5846)
binary-com/deriv-app: packages/cashier/README.md

# `@deriv/cashier`

## Description

This repository is a workspace of [deriv-app](../../README.md) monorepo and covers the [cashier](https://app.deriv.com/cashier/) page in app.deriv.com. (You need to be logged in to see the page.)
This workspace covers:

-   Deposit, Withdraw and Transfer (for both `Fiat` and `Crypto`)
-   Deposit and withdraw with `Payment agents`
-   Deposit with `p2p` and `onRamp`

## How to Install the Project

Follow the instructions [here](../../README.md).

## How to Work with the cashier workspace

To run and work on this workspace you need to use `npm run serve cashier` command along with `npm run serve core`.
Webpack will watch changes in `cashier` so that if you made any changes in this package, it will automatically rebuild `cashier` and recompile `core`.
You can use it like this in your code:

```
const cashier = require('@deriv/cashier');

```

## Folder Structure

```
build
    ├── ...
    ├── webpack-config.js
src
    ├── _common
    ├── components
    ├── config
    ├── constants
    |   |──routes-config.js
    ├── containers
    ├── pages
    ├── public
    ├── stores
    ├── utils

>     │   ├── Validator

Reviewer: ```suggestion
    │   ├── validator
```

# `@deriv/cashier`

## Description

This repository is a workspace of [deriv-app](../../README.md) monorepo and covers the [cashier](https://app.deriv.com/cashier/) page in app.deriv.com. (You need to be logged in to see the page.)

> This project covers:

Reviewer 2: ```suggestion
This workspace covers:
```

-   Deposit, Withdraw and Transfer (for both `Fiat` and `Crypto`)

> -   Deposit and withdraw for `payment agent`

Reviewer 3: ```suggestion
-   Deposit and withdraw with `Payment agents`
```

> -   Deposit for `p2p` and `onRamp`

Reviewer 3: ```suggestion
-   Deposit with `p2p` and `onRamp`
```

## How to Install the Project

Follow the instructions [here](../../README.md).

> ## How to Work with the Project

Reviewer 2: ```suggestion
## How to Work with the cashier workspace
```

To run and work on this workspace you need to use `npm run serve cashier` command along with `npm run serve core`.
Webpack will watch changes in `cashier` so that if you made any changes in this package, it will automatically rebuild `cashier` and recompile `core`.
You can use it like this in your code:

```
const cashier = require('@deriv/cashier');

```

## Folder Structure

```
build
    ├── ...
    ├── webpack-config.js
src
    ├── _common
    ├── components
    ├── config
    ├── constants
    |   |──routes-config.js
    ├── containers
    ├── pages
    ├── public
    ├── stores
    ├── utils
    │   ├── Validator
    |   |   ├──...
    |   |   ├── validator.js

```

> **components:** This folder contains all the reusable components we need for developing the project.

Reviewer 2: ```suggestion
**components:** This folder contains all the reusable components that we need for developing this workspace.
```

We have a separate folder for each component.

> **constants:** We add the static data structures needed for the project here.

Reviewer 2: ```suggestion
**constants:** We add the static data structures needed for the workspace here.
```

> We have a separate folder for each component.

Reviewer 3: This line is redundant.

> **containers:** This folder contains the wrappers we use for the whole project.

Reviewer 2: ```suggestion
**containers:** This folder contains the wrappers we use for the whole workspace.
```

**pages:** This is the place for putting all the features which are visible through the browser. We have separate folder for each feature.

> The features developed in this folder also has been exported through the webpack for the use of other packages (like appstore).

Reviewer 3: ```suggestion
The features developed in this folder also have been exported through the webpack for the use of other packages (like appstore).
```

**public:** All shared files (like .svg files) will be placed here.

**store:** We use Mobx as state management tool in Cashier, and this is the place for putting the relevant store files. Also, for each store file we have a test file in **tests** folder.

> **utils:** We place all the common and helper methods which are required for the project in this folder.

Reviewer 2: ```suggestion
**utils:** We place all the common and helper methods which are required for the workspace in this folder.
```

## Troubleshooting

> -   **Icon missing:** If the icons are missing, you only need to build this project. You can do this by running the build command:

Reviewer 2: ```suggestion
-   **Icon missing:** If the icons are missing, you only need to build this workspace. You can do this by running the build command:
```
